server/src/test/java/org/cloudfoundry/identity/uaa/ServerRunningExtension.java

```diff
@@ -50,44 +47,42 @@
 import java.util.Map;
 
 import static java.nio.charset.StandardCharsets.UTF_8;
-import static org.junit.Assert.assertTrue;
+import static org.assertj.core.api.Assertions.fail;
 
 /**
  * <p>
- * A rule that prevents integration tests from failing if the server application
- * is not running or not accessible. If the server is not running in the
- * background all the tests here will simply be skipped because of a violated
- * assumption (showing as successful). Usage:
+ * An Extension that fails integration tests if the server application
+ * is not running or not accessible.
+ * Usage:
  * </p>
- * 
+ *
  * <pre>
- * &#064;Rule public static ServerRunning brokerIsRunning = ServerRunning.isRunning();
- * 
- * &#064;Test public void testSendAndReceive() throws Exception { // ... test using server etc. }
+ * &#064;RegisterExtension
+ * public static final ServerRunningExtension serverRunning = ServerRunningExtension.connect();
+ *
+ * &#064;Test
+ * void testSendAndReceive() {
+ *      ResponseEntity<Void> response = serverRunning.postForResponse(serverRunning.getAuthorizationUri(), headers, params);
+ * }
  * </pre>
+ *
  * <p>
- * The rule can be declared as static so that it only has to check once for all
- * tests in the enclosing test case, but there isn't a lot of overhead in making
- * it non-static.
+ * The Extension can be declared as static so that it only has to check once for all
+ * tests in the enclosing test case.
```

Comment:
**[question]** I'm not sure this comment makes sense anymore, maybe we should remove it? As I understand it, the check is performed in the `BeforeAll` callback and will run once in any case, whether your register the extension as a static `@RegisterExtension` field or with `@ExtendWith`.